editor/gui/inspector_dock.cpp

```diff
@@ -68,8 +68,15 @@ void InspectorDock::drawSceneActions(Scene* scene)
 			ImGui::SetItemDefaultFocus();
 			ImGui::InputText("##SearchComponent", &searchString);
 
+			Vector<String> components;
 			for (const auto& [componentName, componentData] : ECSFactory::s_ComponentSets)
 			{
+				components.push_back(componentName);
+			}
+			std::sort(components.begin(), components.end());
+			for (const auto& componentName : components)
+			{
+				auto& componentData = ECSFactory::s_ComponentSets[componentName];
```

Comment:
this sort is taking place at every frame, change it to once per execution

Reply:
Fixed